dandiapi/api/views/webdav/__init__.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from django.shortcuts import get_object_or_404
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework.decorators import api_view
+from rest_framework.exceptions import PermissionDenied
+
+from dandiapi.api.asset_paths import get_path_children
+from dandiapi.api.models.asset_paths import AssetPath
+from dandiapi.api.models.version import Version
+from dandiapi.api.services.permissions.dandiset import is_dandiset_owner
+from dandiapi.api.views.common import PAGINATION_PARAMS
+from dandiapi.api.views.pagination import DandiPagination
+
+from .serializers import AtPathQuerySerializer, PathResultSerializer
+
+if TYPE_CHECKING:
+    from django.db.models import QuerySet
+    from rest_framework.request import Request
+
+
+def get_atpath_queryset(*, version: Version, path: str, children: bool) -> QuerySet[AssetPath]:
+    select_related_clauses = ('asset', 'asset__blob')
+    qs = (
+        AssetPath.objects.select_related(*select_related_clauses)
+        .filter(version=version)
+        .order_by('path')
+    )
+
+    # Handle root path case explicitly
+    if path == '':
+        return qs.exclude(path__contains='/') if children else qs.none()
+
+    # Perform path filter
+    qs = qs.filter(path=path)
+
+    # Ensure queryset isn't empty
+    asset_path = qs.first()
+    if asset_path is None:
+        return qs.none()
+
+    # Since path+version combinations are unique, we know we've matched exactly one path.
+    # Now see if we should extend this with it's children
+    if asset_path.asset is not None or not children:
+        return qs
+
+    # Now we know path is a folder, and we should show its children
+    children_paths = (
+        get_path_children(asset_path).select_related(None).select_related(*select_related_clauses)
+    )
+
+    # Must include an additional order clause here to ensure correct ordering
+    return qs.union(children_paths).order_by('path')
+
+
+@swagger_auto_schema(
+    query_serializer=AtPathQuerySerializer,
+    manual_parameters=PAGINATION_PARAMS,
+    responses={200: PathResultSerializer(many=True)},
+    method='GET',
+)
+@api_view(['GET'])
+def atpath(request: Request):
+    query_serializer = AtPathQuerySerializer(data=request.query_params)
+    query_serializer.is_valid(raise_exception=True)
+
+    params = query_serializer.validated_data
+    version = get_object_or_404(
+        Version.objects.select_related('dandiset'),
+        dandiset_id=int(params['dandiset_id']),
+        version=params['version_id'],
+    )
+
+    if not is_dandiset_owner(version.dandiset, request.user):
```

Comment:
How does this work for unauthenticated requests?  They need to be supported, as that's what `dandidav` uses.

Reply:
`dandidav` only serves Dandisets that it can see via unauthenticated requests, which should be the OPEN ones, yes.